### PR TITLE
feat: add CTGAN model

### PR DESCRIPTION
**Pull request by ricardodcpereira**

### CODE REVIEW

File: src/ydata_synthetic/synthesizers/loss.py [hunk 1]
```diff
@@ -1,53 +1,70 @@
-from tensorflow import random
-from tensorflow import reshape, shape, math, GradientTape, reduce_mean
-from tensorflow import norm as tfnorm
+import tensorflow as tf
```
Comment (aquemy) [inline src/ydata_synthetic/synthesizers/loss.py:1]: Can we revert to what was done previous, i.e. importing only what is necessary,

File: src/ydata_synthetic/synthesizers/regular/ctgan/model.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+from joblib import dump
+import numpy as np
+from pandas import DataFrame
+import tensorflow as tf
+from keras.layers import \
+    (Input, Dense, LeakyReLU, Dropout, BatchNormalization, ReLU, Concatenate)
+from keras import Model
+
+import tensorflow_probability as tfp
+from .utils import ConditionalLoss, RealDataSampler, ConditionalSampler
```
Comment (aquemy) [inline src/ydata_synthetic/synthesizers/regular/ctgan/model.py:11]: Please, use only absolute import path.

File: src/ydata_synthetic/synthesizers/regular/model.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 from ..regular.cwgangp.model import CWGANGP
 from ..regular.cramergan.model import CRAMERGAN
 from ..regular.dragan.model import DRAGAN
+from ..regular.ctgan.model import CTGAN
```
Comment (aquemy) [inline src/ydata_synthetic/synthesizers/regular/model.py:17]: Can we change as well these imports for absolute ones please?

**Review by aquemy — CHANGES_REQUESTED** [review]
Very minor comments.

File: src/ydata_synthetic/synthesizers/loss.py (hunk 1 above)
Comment (ricardodcpereira) [reply src/ydata_synthetic/synthesizers/loss.py]: Done.

File: src/ydata_synthetic/synthesizers/regular/model.py (hunk 3 above)
Comment (ricardodcpereira) [reply src/ydata_synthetic/synthesizers/regular/model.py]: Done.

**Review by aquemy — APPROVED** [review]
